pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -97,5 +119,253 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
+	// Check if the MasterUserRecord already exists
+	mur := &toolchainv1alpha1.MasterUserRecord{}
+	// We can use the same NamespacedName from the request as the MasterUserRecord will have the same name as the
+	// UserSignup (i.e. the user's username)
+	err = r.client.Get(context.TODO(), request.NamespacedName, mur)
+	if err != nil {
+		// We generally EXPECT the MasterUserRecord to not be found here, so we only deal with other error types
+		if !errors.IsNotFound(err) {
+			// Error reading the MasterUserRecord, requeue the request
+			return reconcile.Result{}, err
+		}
+	} else {
+		// If we successfully found an existing MasterUserRecord then our work here is done, set the status
+		// to Complete and return
+		statusError := r.setStatusComplete(instance, "")
+		if statusError != nil {
+			reqLogger.Error(statusError, "Error updating UserSignup Status to Complete", "Name", instance.Name)
+			return reconcile.Result{}, statusError
+		}
+		return reconcile.Result{}, nil
+	}
+
+	// Check the user approval policy.
+	userApprovalPolicy, err := r.ReadUserApprovalPolicyConfig()
+	if err != nil {
+		statusError := r.setStatusFailedToReadUserApprovalPolicy(instance, err.Error())
+		if statusError != nil {
+			reqLogger.Error(statusError, "Error updating UserSignup Status", "Name", instance.Name)
+			return reconcile.Result{}, statusError
+		}
+
+		reqLogger.Error(err, "Error reading user approval policy")
```

Comment:
Let's log the error before updating the status. So if the status update failed we will get the original error in the log too.

Reply:
Fixed.